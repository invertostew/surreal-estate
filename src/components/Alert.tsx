import React from "react";

interface AlertProps {
  message: string;
  success?: boolean;
}

const Alert: React.FC<AlertProps> = ({
  message,
  success,
}): JSX.Element | null => {
  if (!message) return null;

  return (
    <section className={`alert alert--${success ? "success" : "error"}`}>
      {message}
    </section>
  );
};

Alert.defaultProps = {
  success: false,
};

export default Alert;
