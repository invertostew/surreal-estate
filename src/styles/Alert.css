.alert {
  box-shadow: 0 4px 6px -1px rgb(0 0 0 / 10%), 0 2px 4px -1px rgb(0 0 0 / 6%);
  padding: 1rem;
  background-color: #fff;
  border: 2px solid #ddd;
  border-radius: 0.75rem;
  margin-bottom: 2rem;
  font-weight: 700;
  text-align: center;
}

.alert--success {
  color: #15803d;
  background-color: #dcfce7;
  border-color: #dcfce7;
}

.alert--error {
  color: #b91c1c;
  background-color: #fee2e2;
  border-color: #fee2e2
}
