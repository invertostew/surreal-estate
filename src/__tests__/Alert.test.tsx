import React from "react";
import { render, screen } from "@testing-library/react";

import Alert from "../components/Alert";

describe("Alert", () => {
  const validProps = {
    message: "",
    success: false,
  };

  it("doesn't render if the message is falsy", () => {
    const { asFragment } = render(
      <Alert message={validProps.message} success={validProps.success} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders a success message if success is true", () => {
    validProps.message = "Success Message";
    validProps.success = true;

    const { asFragment } = render(
      <Alert message={validProps.message} success={validProps.success} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders an error message if success is false", () => {
    validProps.message = "Error Message";
    validProps.success = false;

    const { asFragment } = render(
      <Alert message={validProps.message} success={validProps.success} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders success case in the dom with correct class", () => {
    validProps.message = "Success Message";
    validProps.success = true;

    render(<Alert message={validProps.message} success={validProps.success} />);

    const successAlert = screen.getByText(/success message/i);

    expect(successAlert).toBeInTheDocument();
    expect(successAlert).toHaveClass("alert--success");
  });

  it("renders error case in the dom with correct class", () => {
    validProps.message = "Error Message";
    validProps.success = false;

    render(<Alert message={validProps.message} success={validProps.success} />);

    const errorAlert = screen.getByText(/error message/i);

    expect(errorAlert).toBeInTheDocument();
    expect(errorAlert).toHaveClass("alert--error");
  });
});
