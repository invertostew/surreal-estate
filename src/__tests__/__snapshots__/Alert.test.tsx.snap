// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert doesn't render if the message is falsy 1`] = `<DocumentFragment />`;

exports[`Alert renders a success message if success is true 1`] = `
<DocumentFragment>
  <section
    class="alert alert--success"
  >
    Success Message
  </section>
</DocumentFragment>
`;

exports[`Alert renders an error message if success is false 1`] = `
<DocumentFragment>
  <section
    class="alert alert--error"
  >
    Error Message
  </section>
</DocumentFragment>
`;
